File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/DiyabetFormViewModel.ts
//$2D405E0B
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { DiyabetVeriSeti } from 'NebulaClient/Model/AtlasClientModel';
import { DiyabetKompBilgisi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSDiyabetEgitimi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSDiyabetKomplikasyonlari } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';

export class DiyabetFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.DiyabetFormViewModel, Core';
    @Type(() => DiyabetVeriSeti)
    public _DiyabetVeriSeti: DiyabetVeriSeti = new DiyabetVeriSeti();
    @Type(() => DiyabetKompBilgisi)
    public DiyabetKompBilgisiGridList: Array<DiyabetKompBilgisi> = new Array<DiyabetKompBilgisi>();
    @Type(() => SKRSDiyabetEgitimi)
    public SKRSDiyabetEgitimis: Array<SKRSDiyabetEgitimi> = new Array<SKRSDiyabetEgitimi>();
    @Type(() => SKRSDiyabetKomplikasyonlari)
    public SKRSDiyabetKomplikasyonlaris: Array<SKRSDiyabetKomplikasyonlari> = new Array<SKRSDiyabetKomplikasyonlari>();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/DiyabetFormViewModel.cs
//$5A397872
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class DiyabetVeriSetiServiceController
    {
        partial void PreScript_DiyabetForm(DiyabetFormViewModel viewModel, DiyabetVeriSeti diyabetVeriSeti, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._DiyabetVeriSeti.EpisodeAction = episodeAction;
                viewModel._DiyabetVeriSeti.SubEpisode = episodeAction.SubEpisode;

            }
            ContextToViewModel(viewModel, objectContext);
        }
    }
}

namespace Core.Models
{
    public partial class DiyabetFormViewModel: BaseViewModel,IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._DiyabetVeriSeti);

            if (this.DiyabetKompBilgisiGridList != null)
            {
                foreach (var item in this.DiyabetKompBilgisiGridList)
                {
                    var diyabetKompBilgisiGridListImported = (DiyabetKompBilgisi)objectContext.AddObject(item);
                    diyabetKompBilgisiGridListImported.DiyabetVeriSeti = this._DiyabetVeriSeti;
                }
            }

            var diyabet = (DiyabetVeriSeti)objectContext.GetLoadedObject(this._DiyabetVeriSeti.ObjectID);


            new SendToENabiz(objectContext, diyabet.SubEpisode, diyabet.ObjectID, diyabet.ObjectDef.Name, "215", Common.RecTime());
        }
    }
}
